### ISSUE TITLE
[4.x]: Query Aborted Exception in query when search results empty

### ISSUE THREAD

**Issue by dgsiegel** [craftcms/cms#14623]
### Description

Similarly to https://github.com/craftcms/cms/issues/14552 we get a `Query Aborted Exception` when running a search query that returns no results, ie. `craft.entries().search('foobar-does-not-exist')`.

### Steps to reproduce

1. Search for something that does not exist on your site

### Expected behavior

Search should return 0 matches.

4.8.1 works fine. The culprit might just be https://github.com/craftcms/cms/commit/01a1638395f5e2518c854f973d9bc4adb560afa2

### Craft CMS version

4.8.3

### PHP version

8.2

### Operating system and version

Debian

### Database type and version

MariaDB

### Installed plugins and versions

-

**Comment by i-just** [comment]
Thanks for raising a new issue! I can trigger the `QueryAbortedException` with the following code:
```
{% set query = craft.entries().search('doesnt-exist').orderBy('score') %}
```
If I was to then do `{{ dump(query.rawSql) }}` in my template, I will see the exception. If I do `query.all()`, expected empty array is returned.

What happens is: the exception is thrown here: https://github.com/craftcms/cms/blob/develop/src/elements/db/ElementQuery.php#L2815. (This has been the case since at least version 3.0.0.)

It's then caught via e.g. https://github.com/craftcms/cms/blob/develop/src/db/Query.php#L244-L251 (same applies for other methods, line e.g. `one()`) and empty result is returned.

So, just to clarify, what do you do after `craft.entries().search('foobar-does-not-exist')`? Are you trying to dump the `rawSql`, are you iterating through the results, something else?

**Comment by dgsiegel** [comment]
> So, just to clarify, what do you do after `craft.entries().search('foobar-does-not-exist')`? Are you trying to dump the `rawSql`, are you iterating through the results, something else?

Oh yeah, sorry I didn't mention this earlier. I paginate the results via `{% paginate query as pageInfo, pageEntries %}.` 

Interestingly I can't reproduce the error after clearing all caches, and partially re-creating the search index. So it might just have been a caching/deployment issue.

**Comment by brandonkelly** [comment]
@dgsiegel Can you search for `Query Aborted Exception` / `QueryAbortedException` in your `storage/logs/` folder, and post the stack trace that follows it?

**Comment by dgsiegel** [comment]
> @dgsiegel Can you search for `Query Aborted Exception` / `QueryAbortedException` in your `storage/logs/` folder, and post the stack trace that follows it?

Sure thing. Let me know if you need more details from the log:

```
2024-03-19 10:40:28 [web.ERROR] [craft\db\QueryAbortedException] craft\db\QueryAbortedException in /var/www/[redacted]/vendor/craftcms/cms/src/elements/db/ElementQuery.php:2851
Stack trace:
#0 /var/www/[redacted]/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1501): craft\elements\db\ElementQuery->_applySearchParam()
#1 /var/www/[redacted]/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1833): craft\elements\db\ElementQuery->prepare()
#2 /var/www/[redacted]/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1580): craft\elements\db\ElementQuery->prepareSubquery()
#3 /var/www/[redacted]/vendor/craftcms/cms/src/db/Paginator.php(135): craft\elements\db\ElementQuery->count()
#4 /var/www/[redacted]/vendor/craftcms/cms/src/db/Paginator.php(158): craft\db\Paginator->getTotalResults()
#5 /var/www/[redacted]/vendor/craftcms/cms/src/db/Paginator.php(180): craft\db\Paginator->getTotalPages()
#6 /var/www/[redacted]/vendor/craftcms/cms/src/db/Paginator.php(103): craft\db\Paginator->setCurrentPage()
#7 /var/www/[redacted]/vendor/craftcms/cms/src/helpers/Template.php(161): craft\db\Paginator->__construct()
#8 /var/www/[redacted]/storage/runtime/compiled_templates/35/35e598deed22afc49a5742cca007844c.php(114): craft\helpers\Template::paginateQuery()
#9 /var/www/[redacted]/vendor/twig/twig/src/Template.php(394): __TwigTemplate_6fcd81164cfce23bfc37ef9c241f992a->doDisplay()
#10 /var/www/[redacted]/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling()
#11 /var/www/[redacted]/vendor/twig/twig/src/Template.php(379): Twig\Template->display()
#12 /var/www/[redacted]/vendor/twig/twig/src/TemplateWrapper.php(38): Twig\Template->render()
#13 /var/www/[redacted]/vendor/twig/twig/src/Environment.php(280): Twig\TemplateWrapper->render()
#14 /var/www/[redacted]/vendor/craftcms/cms/src/web/View.php(482): Twig\Environment->render()
#15 /var/www/[redacted]/vendor/craftcms/cms/src/web/View.php(535): craft\web\View->renderTemplate()
#16 /var/www/[redacted]/vendor/craftcms/cms/src/web/TemplateResponseFormatter.php(57): craft\web\View->renderPageTemplate()
#17 /var/www/[redacted]/vendor/yiisoft/yii2/web/Response.php(1100): craft\web\TemplateResponseFormatter->format()
#18 /var/www/[redacted]/vendor/craftcms/cms/src/web/Response.php(337): yii\web\Response->prepare()
#19 /var/www/[redacted]/vendor/yiisoft/yii2/web/Response.php(340): craft\web\Response->prepare()
#20 /var/www/[redacted]/vendor/yiisoft/yii2/base/Application.php(390): yii\web\Response->send()
#21 /var/www/[redacted]/htdocs/index.php(12): yii\base\Application->run()
#22 {main} {"memory":5993472,"exception":"[object] (craft\\db\\QueryAbortedException(code: 0):  at /var/www/[redacted]/vendor/craftcms/cms/src/elements/db/ElementQuery.php:2851)"}
```

**Comment by brandonkelly** [comment]
Based on your stack trace it looks like you’re running Craft from the 4.9 branch, not 4.8.3. And you’re a couple commits behind, possibly 0ff4796abd9d073d8c3d15a934e381f90cb403fc. This particular bug was already fixed shortly after that commit, via e49c617ebbbdd3002ccf3f352cdafb8bec990976.

In the future, please be careful to note the actual Craft version you’re running, and if you’re on a development branch, first run `composer update` to ensure you’re up to date (development branches are not stable and change rapidly), and note the specific commit hash you’re running.

You can get the current commit hash by running the following command:

```sh
composer show craftcms/cms | grep dist
```
